src/reader.rs

```diff
@@ -179,7 +179,7 @@ impl ReaderOffset for usize {
 ///
 /// All read operations advance the section offset of the reader
 /// unless specified otherwise.
-pub trait Reader: Debug + Clone {
+pub trait Reader: Debug + Clone + Send + Sync {
```

Comment:
I'm not sure about this (and same for `Endianity`). Nothing in gimli requires this. I think this should be up to the consumer to decide, but this isn't something I've had much experience with. We're already defining a `Reader` trait in dwarfdump that could specify this requirement for its use.

Reply:
Unfortunately this is tricky because there doesn't seem to be a way for `dwarfdump::Reader` to constrain its `Endian` to be `Send + Sync`. This is related to https://github.com/rust-lang/rust/issues/38738.

Reply:
I figured something out.